applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -757,7 +777,7 @@ static bool handle_hid_report_event(const struct hid_report_event *event)
 		 * avoid negatively affecting HID report rate.
 		 */
 		scan_start_counter = 0;
-	} else if ((state == STATE_ACTIVE) || (state == STATE_DELAYED_ACTIVE)) {
+	} else if (state == STATE_ACTIVE) {
```

Comment:
this change looks like a side fix, it would be better to move it to the commit that introduces the state machine.

Reply:
Actually it's a proper update that is needed after `STATE_FORCED_ACTIVE` was introduced. The `STATE_DELAYED_ACTIVE` is going to become `STATE_FORCED_ACTIVE` so that we should not disable scanning on HID report when in `STATE_DELAYED_ACTIVE` state. Previously, the `STATE_DELAYED_ACTIVE` used to become `STATE_ACTIVE`, so proper behavior here was to stop scanning on HID report event.

Reply:
I see, thanks